flink-runtime/src/main/java/org/apache/flink/runtime/rest/AbstractHandler.java

```diff
@@ -84,8 +84,8 @@ protected AbstractHandler(
 
 	@Override
 	protected void respondAsLeader(ChannelHandlerContext ctx, Routed routed, T gateway) throws Exception {
-		if (log.isDebugEnabled()) {
-			log.debug("Received request " + routed.request().getUri() + '.');
+		if (log.isTraceEnabled()) {
```

Comment:
Does `routed.request().getUri()` perform some decoding work? If not, why not simply use
```java
log.trace("Received request {}.", routed.request().getUri());
```
which should be very efficient (no object creation, string concatenation, etc). Results even in less byte code.

Reply:
You are right. Keep it as is, to be safe.